core/src/main/java/org/apache/calcite/rel/metadata/RelMdColumnUniqueness.java

```diff
@@ -478,17 +525,78 @@ private static ImmutableBitSet decorateWithConstantColumnsFromPredicates(
       ImmutableBitSet checkingColumns, RelNode rel, RelMetadataQuery mq) {
     final RelOptPredicateList predicates = mq.getPulledUpPredicates(rel);
     if (!RelOptPredicateList.isEmpty(predicates)) {
-      final Set<Integer> constantIndexes = new HashSet<>();
-      predicates.constantMap.keySet().forEach(rex -> {
-        if (rex instanceof RexInputRef) {
-          constantIndexes.add(((RexInputRef) rex).getIndex());
-        }
-      });
+      ImmutableBitSet constantIndexes = getConstantColumnSet(predicates);
       if (!constantIndexes.isEmpty()) {
         return checkingColumns.union(ImmutableBitSet.of(constantIndexes));
       }
     }
     // If no constant columns deduced, return the original "checkingColumns".
     return checkingColumns;
   }
+
+  /**
+   * Return the set of columns that are set to a constant literal or a scalar query (as
+   * in a correlated subquery). Examples of constants are {@code x} in the following:
+   * <pre>SELECT x FROM table WHERE x = 5</pre>
+   * and
+   * <pre>SELECT x, y FROM table WHERE x = (SELECT MAX(x) FROM table)</pre>
+   * <p/>
```

Comment:
Remove `</p>`. We don't use end-of-paragraph markers.

Add `<p>` at start of line, after blank line.